src/agent/src/agent.cpp

```diff
@@ -28,6 +29,8 @@ Agent::~Agent()
 
 void Agent::Run()
 {
+    Configuration config;
```

Comment:
You should use the `ConfigurationParser` to read the modules configuration. Check how we do with the `communicator` here:
https://github.com/wazuh/wazuh-agent/pull/45/files#diff-da3d5e8dd4752020c509b5b337eb7254314aad465439e743138acdb52912a34cR20